src/Uno.UI/KeyboardRectProvider.Android.cs

```diff
@@ -66,25 +68,28 @@ public void Stop()
 		/// </summary>
 		void ViewTreeObserver.IOnGlobalLayoutListener.OnGlobalLayout()
 		{
-			var metrics = new DisplayMetrics();
-			_activity.WindowManager.DefaultDisplay.GetMetrics(metrics);
+			var realMetrics = Get<DisplayMetrics>(_activity.WindowManager.DefaultDisplay.GetRealMetrics);
+			var displayRect = Get<Rect>(_activity.WindowManager.DefaultDisplay.GetRectSize);
+			var usableRect = Get<Rect>(_popupView.GetWindowVisibleDisplayFrame);
 
-			var realMetrics = new DisplayMetrics();
-			_activity.WindowManager.DefaultDisplay.GetRealMetrics(realMetrics);
+			// we assume that the keyboard and the navigation bar always occupy the bottom area, with the keyboard being above the navigation bar
```

Comment:
This comment made me think about the floating keyboards on samsung phones, I wonder how it behaves.

Reply:
It looks like an overlay to me. Given that "You can move it and re-size it with intuitive drag and drop", I doubt it would be taken into account for `DefaultDisplay.GetRectSize`.

*Needs to be confirmed:*
> Likely we will get an empty rect for keyboard here, resulting in `InputPaneVisible` being false. This would in turn disable the "scroll into focused element" behavior, 

... which should be the expected behavior since that the user is in control of the overlay keyboard.

Reply:
Confirmed:
![screencapture-file-c-users-xiaotian-gu-downloads-asd-html-2018-11-22-12_41_40](https://user-images.githubusercontent.com/32882405/48917305-1c0bd080-ee54-11e8-9ddf-f5b604151c6e.png)

It is strange that on samsung S8+ with android 7.0, `displayRect` have the space for nav bar reserved even when it is hidden. However, `occupiedRect` still stands correct.